### PR TITLE
feat(Otel): create first version of Otel interface

### PR DESCRIPTION
**Pull request by fgozdz**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
   "dependencies": {
     "@commitlint/cli": "^19.4.1",
```
Comment (manast) [inline package.json:36]: all @commitlint and @semantic-release packages are "devDependencies".

File: src/otel.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    getTracer(name: string, version?: string): Tracer {
+        return this.traceAPI.getTracer(name, version);
+    }
+
+    setSpan(context: OtelContext, span: OtelSpan): OtelContext {
+        return this.traceAPI.setSpan(context, span);
+    }
+}
+
+export class Otel implements Telemetry<OtelContext> {
```
Comment (manast) [inline src/otel.ts]: Rename ```Otel``` to ```BullMQOTel```

File: src/otel.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    with<A extends (...args: any[]) => any>(context: OtelContext, fn: A): ReturnType<A> {
+        return this.contextAPI.with(context, fn);
+    }
+
+    active(): OtelContext {
+        return this.contextAPI.active();
+    }
+}
+
+class ExtendedTrace implements Trace<OtelSpan> {
```
Comment (manast) [inline src/otel.ts]: Rename to ```OTelTrace```

File: src/otel.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import { TraceAPI, trace as OtelTrace, PropagationAPI, ContextAPI, context, propagation, Context as OtelContext, Span as OtelSpan } from '@opentelemetry/api';
+import { Carrier, ContextManager, Telemetry, Trace, Tracer } from 'bullmq';
+
+class ExtendedContextManager implements ContextManager<OtelContext> {
```
Comment (manast) [inline src/otel.ts]: Rename to ```OTelContextManager```

File: src/otel.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    private contextAPI: ContextAPI;
+
+    constructor() {
+        this.contextAPI = context;
```
Comment (manast) [inline src/otel.ts]: This assignment is not needed as you can just use context directly.

File: src/otel.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    getMetadata(context: OtelContext) {
```
Comment (manast) [inline src/otel.ts]: rename to context to ```ctx``` so that it does not collide with the imported ```context``` variable.

File: src/otel.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    active(): OtelContext {
+        return this.contextAPI.active();
+    }
+}
+
+class ExtendedTrace implements Trace<OtelSpan> {
+    private traceAPI: TraceAPI;
+
+    constructor() {
+        this.traceAPI = OtelTrace;
```
Comment (manast) [inline src/otel.ts]: This assignment and private ```traceAPI``` are not needed as you can use ```OTelTrace``` directly.

**Review by manast — COMMENTED** [review]
Simple and nice, just a few comments.

File: package.json [hunk 8]
```diff
@@ -26,18 +26,20 @@
   },
   "homepage": "https://github.com/taskforcesh/bullmq-otel#readme",
   "dependencies": {
+    "@opentelemetry/api": "^1.9.0",
+    "bullmq": "^5.12.12",
+    "typedoc": "^0.26.6",
```
Comment (roggervalf) [inline package.json]: this should be a dev-dependency as well, along with typescript

**Review by manast — APPROVED** [review]
LGTM

**Comment by manast** [comment]
I had to install tslib in order to compile the library: 
```bash
npm install --save-dev tslib
```
Otherwise I got this error: 
```
src/index.ts:1:1 - error TS2354: This syntax requires an imported helper but module 'tslib' cannot be found.

1 export * from './bullMQOtel';
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in src/index.ts:1``

**Pull request merged**